packages/mui-material-next/src/Button/Button.tsx

```diff
@@ -469,6 +469,8 @@ const Button = React.forwardRef(function Button<
     tabIndex,
     type,
     variant,
+    // required for making sure the sx prop is handled correctly
+    materialYouComponent: true,
```

Comment:
I still don't understand why this flag is needed.

Reply:
Can you give some examples?

Reply:
Sure, MD2 and MD3 components have different mappings for the sx prop (they basically need different sx config which are defined in their respective themes). However, if we want these two types of components to live on the same page (under the same theme provider), which is likely what people would try first, we need to be able to somehow differentiate the two when processing the `sx` prop. For example, if the `sx` prop is used on MD2 component, use the sx config for MD2, and if the component is MD3 component, use the other config. We are making this decision in:

```js
  // in the Material You theme 
  theme.unstable_calculateSxConfig = (props) => {
    if (props.ownerState?.materialYouComponent) {
      return theme.unstable_sxConfig;
    }
    return systemDefaultSxConfig;
  };
```

This is a theme callback that if exists will be invoked inside the `styleFunctionSx` when getting the sx config value from the theme.

Reply:
I've pushed another commit, but honestly I think even that solution would be problematic, especially with the `theme.unstable_sx` prop, as you can't provide additional props for the component (like the `materialYouComponent` flag). We'll need to find a better solution here. I am running out of ideas.

Reply:
I'd say that we should remove the flag for now. Think of Material You as a separate library similar to Joy UI. Then add a doc similar to https://mui.com/joy-ui/guides/using-joy-ui-and-material-ui-together/ that guides how to use MD3 in the existing MD2 project.

In the doc, we can add a custom `sxConfig` snippet that makes the `sx` prop works for both packages.

Does this make sense?

Reply:
Tested on my machine, this works without the flag:

```js
import * as React from 'react';
import { getPath } from '@mui/system';
import { CssVarsProvider, extendTheme } from '@mui/material-next/styles';
import Button from '@mui/material-next/Button';
import Md2Button from '@mui/material/Button';

const customTheme = extendTheme({
  unstable_sxConfig: {
    bgcolor: {
      style: ({ theme, bgcolor }) => {
        return {
          backgroundColor:
            getPath(theme, `sys.color.${bgcolor}`, true) ||
            getPath(theme, `ref.palette.${bgcolor}`, true) ||
            getPath(theme, `palette.${bgcolor}`, true) ||
            bgcolor,
        };
      },
    },
  },
});

export default function Md3() {
  return (
    <CssVarsProvider theme={customTheme}>
      <Button
        variant="filled"
        sx={(theme) => {
          console.log('theme', theme);
          return {};
        }}
      >
        Test
      </Button>
      <Md2Button sx={{ bgcolor: 'onPrimary' }}>Test</Md2Button>
    </CssVarsProvider>
  );
}
```

Reply:
Actually you are right, this would work as long as there are no conflicts in the tokens. Luckily in MD2 the main tokens are  main, light, dark. Let me check if this applies with all sx tokens

Reply:
Done

Reply:
> Think of Material You as a separate library similar to Joy UI. Then add a doc similar to https://mui.com/joy-ui/guides/using-joy-ui-and-material-ui-together/ that guides how to use MD3 in the existing MD2 project.

I have tried the implications of this PR in https://codesandbox.io/s/material-ui-feat-joy-ui-forked-z9bv90?file=/demo.tsx. It seems to mostly work like before. It's a bit sad though that we can't have a different emotion context. I had a friend who recently asked me about using Joy UI inside an existing Material UI app (an engineer without a front-end background). He was intimidated by the size of the example, he didn't try 🙃.